asm_to_pil/src/vm_to_constrained.rs

```diff
@@ -413,45 +419,41 @@ impl<T: FieldElement> ASMPILConverter<T> {
     fn handle_assignment(
         &mut self,
         _start: usize,
-        write_regs: Vec<String>,
-        assign_reg: Option<String>,
+        lhs_with_reg: Vec<(String, String)>,
         value: Expression<T>,
     ) -> CodeLine<T> {
-        assert!(write_regs.len() <= 1);
-        assert!(
-            assign_reg.is_some(),
-            "Implicit assign register not yet supported."
-        );
-        let assign_reg = assign_reg.unwrap();
+        assert!(lhs_with_reg.len() == 1);
```

Comment:
why 1?

Reply:
Multi-assignments are only implemented for function calls (in `self.handle_functional_instruction`). `handle_assignment` only gets called for assignments where the RHS is not a function call, see line 230.

I renamed `handle_assignment` -> `handle_non_functional_assignment` and added an error message to make this clearer.